### PR TITLE
Supervisor Subject Show

### PR DESCRIPTION
**Pull request by shakil-fg**

### CODE REVIEW

File: app/views/supervisor/subjects/show.html.erb [hunk 1]
```diff
@@ -1,2 +1,85 @@
...
+
+<div class="row">
+  <div class="col-sm-8">
+    <%= link_to supervisor_subject_path(@subject), method: :delete, data: { confirm: (t :remove_confirmation) } do %>
+      <button type="button" class="btn btn-lg btn-danger pull-right"><%= t :delete %></button>
+    <% end %>
+    <%= link_to edit_supervisor_subject_path(@subject) do %>
+      <button type="button" class="btn btn-lg btn-default pull-right"><%= t :edit %></button>
+    <% end %>
+
```
Comment (mdmahmudurrahman) [inline app/views/supervisor/subjects/show.html.erb]: Please remove the new line from here

File: app/views/supervisor/subjects/show.html.erb [hunk 2]
```diff
@@ -1,2 +1,85 @@
...
+            <%= link_to '#' do %> <!-- supervisor_course_path(course) -->
+              <%= course.name %>
+            <% end %>
+          </td>
+          <td>NOT IMPLEMENTED YET</td>
+        </tr>
+      <% end %>
+      </tbody>
+    </table>
+
```
Comment (mdmahmudurrahman) [inline app/views/supervisor/subjects/show.html.erb]: Extra line!

File: app/views/supervisor/subjects/show.html.erb [hunk 3]
```diff
@@ -1,2 +1,85 @@
...
+            <%= link_to '#' do %> <!-- supervisor_user_path(user) -->
+              <%= user.name %>
+            <% end %>
+          </td>
+          <td><%= user.email %></td>
+        </tr>
+      <% end %>
+      </tbody>
+    </table>
+
```
Comment (mdmahmudurrahman) [inline app/views/supervisor/subjects/show.html.erb]: Extra line!

**Comment by manazirahsan** [comment]
Okey

**Comment by mdmahmudurrahman** [comment]
Okey

**Comment by shakil-fg** [comment]
Ok

File: app/models/user.rb [hunk 4]
```diff
@@ -26,4 +26,6 @@ class User < ActiveRecord::Base
       supervisor: "Supervisor"
   }
 
+  scope :supervisors, -> { where role: "supervisor" }
```
Comment (chuanhtuan) [inline app/models/user.rb]: study about enum

File: app/models/user.rb [hunk 5]
```diff
@@ -26,4 +26,8 @@ class User < ActiveRecord::Base
       supervisor: "Supervisor"
   }
 
+  enum role: { supervisor: "Supervisor", trainee: "Trainee" }
+
+  scope :supervisors, -> { where(role: User.roles[:supervisor]) }
```
Comment (chuanhtuan) [inline app/models/user.rb]: remove
when you used enum, you can remove 2 copes :)

**Comment by noor-framgia** [comment]
ok

**Comment by mdmahmudurrahman** [comment]
OK

**Comment by manazirahsan** [comment]
Okey

**Comment by shakil-fg** [comment]
Sir, Enum implementation in rails was interesting. Thank you :)

**Pull request merged**